plat/kvm/arm/setup.c

```diff
@@ -208,6 +212,13 @@ static void _libkvmplat_entry2(void *arg __attribute__((unused)))
 void _libkvmplat_start(void *dtb_pointer)
 {
 	_init_dtb(dtb_pointer);
+
+#ifdef CONFIG_HAVE_SMP
+	int ret = smp_init();
+	if (ret < 0)
+		uk_pr_err("SMP init failed\n");
```

Comment:
I think we have to choices:
- simply exiting if `smp_init()` failed
- set a global variable if `smp_init()` failed and introduce an assertion at the beginning of every SMP function, so that SMP code won't be run if `smp_init()` failed

Reply:
I would vote for the former, as it's more flexible than the latter.

Reply:
For now, I would just exit/crash if SMP initialization failed. This usually means something very basic like configuring the APIC did not work, so it does not make sense to continue running in this unsafe state. We do it the same with paging and many other very basic functionality.

Reply:
ACK, will add an assert.